### ISSUE TITLE
HarmonyOS local file support

### ISSUE THREAD

**Issue by Ultra119** [ProfessorJTJ/HISuite-Proxy#87]
I add local Harmony files, run setUp, but HiSuite doesn't find the new version
![Screenshot_21](https://user-images.githubusercontent.com/64977575/129913882-98d4b2d1-90e0-4c45-8c67-4034f78a73b2.png)
![Screenshot_22](https://user-images.githubusercontent.com/64977575/129914098-a193ec35-5201-4184-851a-bebe5c2b73ef.png)

**Comment by ProfessorJTJ** [comment]
You need to know the roms id and add their links like this:

http://update.dbankcdn.com/TDS/data/files/p3/s15/G5625/g1755/vTHEROMIDSHOULDBEHER3/f1/full/

**Comment by Ultra119** [comment]
> You need to know the roms id and add their links like this:
> 
> http://update.dbankcdn.com/TDS/data/files/p3/s15/G5625/g1755/vTHEROMIDSHOULDBEHERE/f1/full/
I have links. How do I find out the ID?

**Comment by Ultra119** [comment]
Sorry, im stupid

**Comment by ProfessorJTJ** [comment]
> > You need to know the roms id and add their links like this:
> > http://update.dbankcdn.com/TDS/data/files/p3/s15/G5625/g1755/vTHEROMIDSHOULDBEHERE/f1/full/
> 
> I have links. How do I find out the ID?

Bad luck, you can longer find ids from links...
The offline package feature is added to support the old links of Huawei which had the rom id in the link.

HOWEVER I might be able to get the firmwares ids for you...

Want me to do that?

**Comment by Ultra119** [comment]
> > > You need to know the roms id and add their links like this:
> > > http://update.dbankcdn.com/TDS/data/files/p3/s15/G5625/g1755/vTHEROMIDSHOULDBEHERE/f1/full/
> > 
> > 
> > I have links. How do I find out the ID?
> 
> Bad luck, you can longer find ids from links...
> The offline package feature is added to support the old links of Huawei which had the rom id in the link.
> 
> HOWEVER I might be able to get the firmwares ids for you...
> 
> Want me to do that?

Yes, it would be nice 

http://update.dbankcdn.com//download/data/pub_13/HWHOTA_hota_900_9/62/v3/2NcdiKfCTmGjng6T52vzoA/full/update_preload_HLK-AL00_all_cn_R8.zip
http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/21/v3/AJ5E4GH1RTK8wwwDJIahMw/full/update_cust_HLK-AL00_all_cn.zip
http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/34/v3/bSsJaN3NTbuB9wE3Ioxumg/full/update_base.zip

**Comment by ProfessorJTJ** [comment]
> > > > You need to know the roms id and add their links like this:
> > > > http://update.dbankcdn.com/TDS/data/files/p3/s15/G5625/g1755/vTHEROMIDSHOULDBEHERE/f1/full/
> > > 
> > > 
> > > I have links. How do I find out the ID?
> > 
> > 
> > Bad luck, you can longer find ids from links...
> > The offline package feature is added to support the old links of Huawei which had the rom id in the link.
> > HOWEVER I might be able to get the firmwares ids for you...
> > Want me to do that?
> 
> Yes, it would be nice
> 
> http://update.dbankcdn.com//download/data/pub_13/HWHOTA_hota_900_9/62/v3/2NcdiKfCTmGjng6T52vzoA/full/update_preload_HLK-AL00_all_cn_R8.zip
> http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/21/v3/AJ5E4GH1RTK8wwwDJIahMw/full/update_cust_HLK-AL00_all_cn.zip
> http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/34/v3/bSsJaN3NTbuB9wE3Ioxumg/full/update_base.zip

Quick question, how did you get the links?

**Comment by Ultra119** [comment]
> > > > > You need to know the roms id and add their links like this:
> > > > > http://update.dbankcdn.com/TDS/data/files/p3/s15/G5625/g1755/vTHEROMIDSHOULDBEHERE/f1/full/
> > > > 
> > > > 
> > > > I have links. How do I find out the ID?
> > > 
> > > 
> > > Bad luck, you can longer find ids from links...
> > > The offline package feature is added to support the old links of Huawei which had the rom id in the link.
> > > HOWEVER I might be able to get the firmwares ids for you...
> > > Want me to do that?
> > 
> > 
> > Yes, it would be nice
> > http://update.dbankcdn.com//download/data/pub_13/HWHOTA_hota_900_9/62/v3/2NcdiKfCTmGjng6T52vzoA/full/update_preload_HLK-AL00_all_cn_R8.zip
> > http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/21/v3/AJ5E4GH1RTK8wwwDJIahMw/full/update_cust_HLK-AL00_all_cn.zip
> > http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/34/v3/bSsJaN3NTbuB9wE3Ioxumg/full/update_base.zip
> 
> Quick question, how did you get the links?

http sniffer

**Comment by ProfessorJTJ** [comment]
> http sniffer

I've ran a program to search for the id, let's wait for the result....

**Comment by ProfessorJTJ** [comment]
> > http sniffer
> 
> I've ran a program to search for the id, let's wait for the result....

Unfortunately it couldn't find the roms ids.
I still can't figure how you got the links, can I connect to your pc through TeamViewer to see how you do it? ( I mean I know you use a proxy to listen to the requests, but I want to go deeper to see by what request you get the links, that way I also might be able to get the ids.... )

**Comment by ProfessorJTJ** [comment]
> > > > http sniffer
> > > 
> > > 
> > > I've ran a program to search for the id, let's wait for the result....
> > 
> > 
> > Unfortunately it couldn't find the roms ids.
> > I still can't figure how you got the links, can I connect to your pc through TeamViewer to see how you do it? ( I mean I know you use a proxy to listen to the requests, but I want to go deeper to see by what request you get the links, that way I also might be able to get the ids.... )
> 
> By the way, I forgot to write that this is a closed beta.
> As for the sniffer, I did it on my smartphone while downloading the firmware at settings

That's it then, unfortunately due to lack of facilities I cannot test exactly how phones update from within the phone, I could only temper with Hisuite

**Comment by Ultra119** [comment]
> > > > > http sniffer
> > > > 
> > > > 
> > > > I've ran a program to search for the id, let's wait for the result....
> > > 
> > > 
> > > Unfortunately it couldn't find the roms ids.
> > > I still can't figure how you got the links, can I connect to your pc through TeamViewer to see how you do it? ( I mean I know you use a proxy to listen to the requests, but I want to go deeper to see by what request you get the links, that way I also might be able to get the ids.... )
> > 
> > 
> > By the way, I forgot to write that this is a closed beta.
> > As for the sniffer, I did it on my smartphone while downloading the firmware at settings
> 
> That's it then, unfortunately due to lack of facilities I cannot test exactly how phones update from within the phone, I could only tamper with Hisuite

I'll try through a PC through a similar software, I'll write here, if it work

**Comment by ProfessorJTJ** [comment]
> > > > > > http sniffer
> > > > > 
> > > > > 
> > > > > I've ran a program to search for the id, let's wait for the result....
> > > > 
> > > > 
> > > > Unfortunately it couldn't find the roms ids.
> > > > I still can't figure how you got the links, can I connect to your pc through TeamViewer to see how you do it? ( I mean I know you use a proxy to listen to the requests, but I want to go deeper to see by what request you get the links, that way I also might be able to get the ids.... )
> > > 
> > > 
> > > By the way, I forgot to write that this is a closed beta.
> > > As for the sniffer, I did it on my smartphone while downloading the firmware at settings
> > 
> > 
> > That's it then, unfortunately due to lack of facilities I cannot test exactly how phones update from within the phone, I could only tamper with Hisuite
> 
> I'll try through a PC through a similar software, I'll write here, if it work

HiSuite can do beta updates?? 

If so be sure to click on "setup" in HiSuite proxy, it will crack hisuite's http component file and allow you to see even the ssl(https) requests, you will exactly see the rom id and so many other useful information...

**Comment by Ultra119** [comment]
> > > > > > > http sniffer
> > > > > > 
> > > > > > 
> > > > > > I've ran a program to search for the id, let's wait for the result....
> > > > > 
> > > > > 
> > > > > Unfortunately it couldn't find the roms ids.
> > > > > I still can't figure how you got the links, can I connect to your pc through TeamViewer to see how you do it? ( I mean I know you use a proxy to listen to the requests, but I want to go deeper to see by what request you get the links, that way I also might be able to get the ids.... )
> > > > 
> > > > 
> > > > By the way, I forgot to write that this is a closed beta.
> > > > As for the sniffer, I did it on my smartphone while downloading the firmware at settings
> > > 
> > > 
> > > That's it then, unfortunately due to lack of facilities I cannot test exactly how phones update from within the phone, I could only tamper with Hisuite
> > 
> > 
> > I'll try through a PC through a similar software, I'll write here, if it work
> 
> HiSuite can do beta updates??
> 
> If so be sure to click on "setup" in HiSuite proxy, it will crack hisuite's http component file and allow you to see even the ssl(https) requests, you will exactly see the rom id and so many other useful information...
> 
> After you click on "setup" open hisuite's settings and modify the data of the proxy program...

Yes, HiSuite can

**Comment by Ultra119** [comment]
> > > > You need to know the roms id and add their links like this:
> > > > http://update.dbankcdn.com/TDS/data/files/p3/s15/G5625/g1755/vTHEROMIDSHOULDBEHERE/f1/full/
> > > 
> > > 
> > > I have links. How do I find out the ID?
> > 
> > 
> > Bad luck, you can longer find ids from links...
> > The offline package feature is added to support the old links of Huawei which had the rom id in the link.
> > HOWEVER I might be able to get the firmwares ids for you...
> > Want me to do that?
> 
> Yes, it would be nice
> 
> http://update.dbankcdn.com//download/data/pub_13/HWHOTA_hota_900_9/62/v3/2NcdiKfCTmGjng6T52vzoA/full/update_preload_HLK-AL00_all_cn_R8.zip
> http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/21/v3/AJ5E4GH1RTK8wwwDJIahMw/full/update_cust_HLK-AL00_all_cn.zip
> http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/34/v3/bSsJaN3NTbuB9wE3Ioxumg/full/update_base.zip

I just realized that these are not full firmware files. The names must contain the word "full"
Here's a full: Can you check the ID?
http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/d7/v3/UJi7QBlaQyyBHCJSQ6j2Kw/full/update_full_base.zip
http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/c5/v3/Lf7nASC0SnSgjbBTg-NO2g/full/update_full_preload_HLK-AL00_all_cn_R8.zip
https://update.dbankcdn.com/download/data/pub_13/HWHOTA_hota_900_9/a4/v3/ntffs_atQV6VFeCnP9Vwqw/full/update_full_cust_HLK-AL00_all_cn.zip

In short, either I'm stupid, or I didn't understand how to do it... When you enable setup, the update is not available

**Comment by ProfessorJTJ** [comment]
Still the links don't include the ids, look, if you can, install TeamViewer and tell me your id and password, I will help you...

**Comment by Ultra119** [comment]
> Still the links don't include the ids, look, if you can, install TeamViewer and tell me your id and password, I will help you...

let's go to telegram @Ultra119
